src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -0,0 +1,180 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ML.Data
+{
+    public sealed partial class TextLoaderRange
+    {
+        public TextLoaderRange()
+        {
+        }
+
+        /// <summary>
+        /// Convenience constructor for the scalar case.
+        /// Min and Max are set to the single value ordinal.
+        /// When a given column spans only a single column in the
+        /// dataset.
+        /// <see cref="ML.Runtime.Data.TextLoader.Range"/>
+        /// </summary>
+        public TextLoaderRange(int ordinal)
+        {
+
+            Contracts.Assert(ordinal >= 0);
```

Comment:
>Contracts.Assert(ordinal >= 0); [](start = 12, length = 31)

Not asserts! These are public facing methods.

Reply:
More detail: Asserts are debug only constructs, and we only use them to indicate what we *expect* is true -- e.g., verification of invariants in code. These should be `.CheckParam`, etc.

Checks with error messages are more appropriate here. I see your message to @veikko_e@hotmail.com 

---
In reply to: [189048133](https://github.com/dotnet/machinelearning/pull/142#discussion_r189048133) [](ancestors = 189048133)